### PR TITLE
Update hello-world-program.md

### PR DESCRIPTION
**Pull request by darkvallen**
### Problem
- Grammatical mistakes
- The dependencies and client code needed updates to ensure compatibility with the latest versions and best practices.

### Summary of Changes
- Fixed grammatical mistakes
- Rephrase some words to make them sound more fluent and easier to read.
- Added a tutorial video for a visual walkthrough.
- Updated instructions for installing dependencies and running the client code.
- Updated dependencies and client code

Fixes #
https://github.com/Unboxed-Software/solana-hello-world-client/pull/6

### CODE REVIEW

**Comment by mikemaccana** [comment]
Thanks @darkvallen ! I want to let you know there's another, much larger PR for this lesson that's been submitted by someone else already, however we're very grateful for these changes - I'll review this PR shortly.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 1]
```diff
@@ -14,28 +14,26 @@ description:
 ## Summary
 
 - Native Solana programs have a single **entry point** to process instructions
-- A program processes an instruction using the **program_id**, list of
+- A program processes an instruction using the **program_id**, a list of
   **accounts**, and **instruction_data** included with the instruction
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:18]:
```suggestion
- A program processes an instruction using the **program_id**, a set of named
  **accounts**, and **instruction_data** included with the instruction
```

File: content/courses/native-onchain-development/hello-world-program.md [hunk 2]
```diff
@@ -73,7 +71,7 @@ following structure:
 
 The path to `AccountInfo` would be `solana_program::account_info::AccountInfo`.
 
-Absent of any other keywords, we would need to reference this entire path to use
+Absent any other keywords, we would need to reference this entire path to use
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:74]:
```suggestion
Without any `use` keywords, we would need to reference this entire path to use
```

File: content/courses/native-onchain-development/hello-world-program.md [hunk 3]
```diff
@@ -335,38 +333,57 @@ Now let's build and deploy our program using Solana Playground.
 
 #### 5. Invoke Program
 
-Finally, let's invoke our program from the client side. The focus of this lesson
-is to build our Solana program, so we’ve gone ahead and provided
-[the client code to invoke our “Hello, world!” program](https://github.com/Unboxed-Software/solana-hello-world-client)
-for you to download.
+In this final step, we'll invoke the "Hello, world!" Solana program from the client side. While the main focus of this lesson is on building your Solana program, we've provided [the client code](https://github.com/Unboxed-Software/solana-hello-world-client) to help you easily interact with your deployed program.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:336]: Moved to https://github.com/solana-developers/hello-world-client - can you please make your updates there instead and fix these links?

File: content/courses/native-onchain-development/hello-world-program.md [hunk 4]
```diff
@@ -335,38 +333,57 @@ Now let's build and deploy our program using Solana Playground.
...
+
+Before we dive into the steps, watch the tutorial video below for a visual walkthrough:
+
+![hello-world-solana](https://github.com/user-attachments/assets/6865e275-bbb9-43a1-acf7-8a97438c8ed5)
+
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:341]: Video is a good idea. The quality of the video is bad and there seems to be no sound - want to re-record and add a voiceover? 

Then contact me via telegram or twitter and send me the raw video for upload to our YouTube channel.

File: content/courses/offline-transactions/durable-nonces.md [hunk 5]
```diff
@@ -14,39 +14,39 @@ description: "Use durable nonces to sign transactions ahead of time."
   have an expiration date of 150 blocks (~80-90 seconds).
```
Comment (mikemaccana) [inline content/courses/offline-transactions/durable-nonces.md:14]: @darkvallen  you also have https://github.com/solana-foundation/developer-content/pull/450 which is for durable-nonces.md. It makes it **very** difficult for us to track your work when this happens. 

Please put all your changes to durable-nonces.md in that PR and remove it from this PR.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Some good changes but also needs some tweaks - see comments above.

File: content/courses/offline-transactions/durable-nonces.md (hunk 5 above)
Comment (darkvallen) [reply content/courses/offline-transactions/durable-nonces.md]: Fixed.
https://github.com/solana-foundation/developer-content/pull/450 Ready for further review.

**Pull request closed without merging**